### ISSUE TITLE
Error: Cannot find module 'babel/register'

### ISSUE THREAD

**Issue by nfriedly** [yeoman/generator-node#131]
I did a `yo node` with v1.2.0 and then ran `gulp` in the output directory with the flowing result:

```
$ gulp
[11:09:09] Using gulpfile ~/running-average/gulpfile.js
[11:09:09] Starting 'static'...
[11:09:09] Starting 'pre-test'...
[11:09:09] Finished 'pre-test' after 109 ms
[11:09:09] Starting 'test'...
[11:09:09] 'test' errored after 5.69 ms
[11:09:09] Error: Cannot find module 'babel/register'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Array.forEach (native)
    at module.exports (/Users/nfriedly/running-average/node_modules/gulp-mocha/index.js:27:16)
    at Gulp.<anonymous> (/Users/nfriedly/running-average/gulpfile.js:41:11)
    at module.exports (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/index.js:214:10)
    at /Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/index.js:279:18
    at finish (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
    at /Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:52:4
    at f (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/node_modules/end-of-stream/node_modules/once/once.js:17:25)
    at DestroyableTransform.onend (/Users/nfriedly/running-average/node_modules/gulp/node_modules/orchestrator/node_modules/end-of-stream/index.js:31:18)
    at DestroyableTransform.emit (events.js:129:20)
    at /Users/nfriedly/running-average/node_modules/gulp-istanbul/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:965:16
    at process._tickDomainCallback (node.js:381:11)

/Users/nfriedly/running-average/test/index.js
  line 3  col 8  'runningAverage' is defined but never used.

  ⚠  1 warning


events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: JSHint failed for: /Users/nfriedly/running-average/test/index.js
```
That looks like a gulp-mocha issue, but if you [dig in](https://github.com/sindresorhus/gulp-mocha/blob/master/index.js#L27), it's just loading the dependencies that the generated gulp.js config tells it to:

```js
gulp.task('test', ['pre-test'], function (cb) {
  var mochaErr;

  gulp.src('test/**/*.js')
    .pipe(plumber())
    .pipe(mocha({reporter: 'spec', require: ['babel/register']}))
    //...
```
However, `babel` is not directly required by the generated `package.json`:

```js
{
//...
  "devDependencies": {
    "gulp": "^3.6.0",
    "gulp-babel": "^5.1.0",
    "gulp-coveralls": "^0.1.0",
    "gulp-istanbul": "^0.8.1",
    "gulp-jscs": "^1.1.0",
    "gulp-jshint": "^1.5.3",
    "gulp-mocha": "^2.0.0",
    "gulp-plumber": "^1.0.0",
    "jshint-stylish": "^1.0.0"
  },
//...
}
```

(Presumably it's there in `gulp-babel`'s node_modules folder, but that isn't good enough for npm 2.x and I don't think it would be correct to skip it even after npm 3.x goes live.)

**Comment by arthurvr** [comment]
We'll need this fix here too: https://github.com/yeoman/generator-gulp-webapp/commit/31348ff1acc393af507a7b8a192d767cf9737021.

**Comment by arthurvr** [comment]
Fixed. Thanks.

**Comment by nfriedly** [comment]
Whoot, nice turnaround time!

**Comment by SBoudrias** [comment]
Released as 1.2.1

**Comment by nfriedly** [comment]
Awesome!

On a vaguely related note, do you guys have any thoughts on adding an option to output ES5 javascript for publishing on npm and such? I know it isn't too difficult to set up, but I was a bit surprised when I realized that it wasn't built-in to the default generator-node setup. (Or is there a flag to turn it on that I just missed?)

**Comment by arthurvr** [comment]
@nfriedly It's [optional](https://github.com/yeoman/generator-node#options). Just use `--no-babel` if you don't want babel.

**Comment by nfriedly** [comment]
Yea, that's what I ended up doing, but I was asking about an option that lets me write es6/7 but automatically generates es5 (in `out/` maybe) and sets that up to be published to npm? So, more of a `--with-debabeling` sort of option.

**Comment by SBoudrias** [comment]
@arthurvr is Babel activated by default? I think we should do simple es5 by default as es6-7 is a harder publishing process.

**Comment by arthurvr** [comment]
@SBoudrias Yes. I like that, personally. That's what we [plan](https://github.com/yeoman/generator-chromeapp/issues/57) for other generators too.

**Comment by SBoudrias** [comment]
@arthurvr could you look what the current case is for generator-node and send a PR fixing it if it's an issue?

**Comment by arthurvr** [comment]
@SBoudrias What do mean with "the current case"? The publishing process? Or the fact that Babel is on by default?

**Comment by SBoudrias** [comment]
@arthurvr Let me rephrase: can you send a PR making babel an option who's not the default?

**Comment by arthurvr** [comment]
Yes I can, though, why? That's not the approach we're taking with other generators. It's future-proof, doesn't complicate the build more than a simple gulp task and I think it's a good idea to promote using ES6.

**Comment by robbbz** [comment]
Why would anybody possible mind more features, in a backwards compatible way? As said in the other generator by silvenon as well, I don't think it should even be optional.

**Comment by SBoudrias** [comment]
Well, speaking for myself here, I still develop in ES5 and I don't intend to start using babel by default anytime soon.

Maybe it's only my opinion, but the overhead of setting up babel is usually not worth the syntactic sugar it provide me.

ES5 is working fine, I don't see why we should be eager to integrate more tooling.
